### DISCUSSION TITLE
Basechain as a  settlement layer.

### DISCUSSION THREAD

**Discussion by askucher** [ethereum-optimism/developers#591] [Core Protocol]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

We have successfully launched l2 with gasCustomToken on the Ethereum Sepolia Testnet as a settlement layer by utilizing the latest version of the Optimism SDK. We want to know what is needed to replace Ethereum with Base as a settlement layer.

Op stack version 1.9.1
Smart contract version: beta.2

Launched services:

1. op-geth 
2. op-node
3. op-batcher
4. op-proposer  

### Additional Information

Everything works fine after deployment on Sepolia Ethereum Testnet, all l1->l2, l2->l2, l2->l1 tests passed. 

`But we got issues when we use other EVM chains as L1. (anvil as l1) `

Some logs:

op-node
```
5|op-node  | t=2024-09-16T21:55:30+0400 lvl=crit msg="Application failed" message="failed to setup: unable to create the rollup node: failed to init L1: failed to validate the L1 config: incorrect L1 genesis block hash 0xb9f22676616cb24aee99ee1c0dca8147a17d05ddd683c5bfaf640ce433665a5b, expected 0x863f4e67fff2a076574ffedd767976cc919646af08aff9086ea3d2ef023b9d4a"
```

op-geth

```
/root/.pm2/logs/op-geth-error.log last 15 lines:
4|op-geth  | INFO [09-16|21:55:25.217] Initialized transaction indexer          range="entire chain"
4|op-geth  | INFO [09-16|21:55:25.217] Initialising Ethereum protocol           network=42069 dbversion=8
4|op-geth  | INFO [09-16|21:55:25.217] Loaded local transaction journal         transactions=0 dropped=0
4|op-geth  | INFO [09-16|21:55:25.217] Chain post-merge, sync via beacon client
4|op-geth  | INFO [09-16|21:55:25.217] Gasprice oracle is ignoring threshold set threshold=2
4|op-geth  | WARN [09-16|21:55:25.219] Engine API enabled                       protocol=eth
4|op-geth  | INFO [09-16|21:55:25.219] Starting peer-to-peer node               instance=Geth/v1.101315.3-stable-8af19cf2/linux-amd64/go1.22.6
4|op-geth  | INFO [09-16|21:55:25.232] IPC endpoint opened                      url=/home/andrey/Projects/op_stack_deployer/op-geth/datadir/geth.ipc
4|op-geth  | INFO [09-16|21:55:25.233] Loaded JWT secret file                   path=jwt.txt crc32=0xfac33720
4|op-geth  | INFO [09-16|21:55:25.233] HTTP server started                      endpoint=[::]:8545 auth=false prefix= cors=* vhosts=*
4|op-geth  | INFO [09-16|21:55:25.233] WebSocket enabled                        url=ws://[::]:8546
4|op-geth  | INFO [09-16|21:55:25.233] WebSocket enabled                        url=ws://[::]:8551
4|op-geth  | INFO [09-16|21:55:25.233] HTTP server started                      endpoint=[::]:8551 auth=true  prefix= cors=localhost vhosts=*
4|op-geth  | INFO [09-16|21:55:25.233] New local node record                    seq=1,726,509,250,068 id=985e6e990566d673 ip=127.0.0.1 udp=0 tcp=30303
4|op-geth  | INFO [09-16|21:55:25.233] Started P2P networking                   self="enode://cad03566a8cad10eebcbbcf53bc923391ad0d97c357864583413bfc8fd6a5505d248cee9cda90dbd57063f19cafb2b119e55f1ee939044296e406b239f382f17@127.0.0.1:30303?discport=0"

```

op-proposer

```
7|op-propo | t=2024-09-17T02:09:31+0400 lvl=warn msg="failed to dial address, but may connect later" addr=http://localhost:9545
7|op-propo | t=2024-09-17T02:09:33+0400 lvl=warn msg="failed to dial address, but may connect later" addr=http://localhost:9545
7|op-propo | t=2024-09-17T02:09:35+0400 lvl=warn msg="failed to dial address, but may connect later" addr=http://localhost:9545
7|op-propo | t=2024-09-17T02:09:37+0400 lvl=warn msg="failed to dial address, but may connect later" addr=http://localhost:9545
7|op-propo | t=2024-09-17T02:09:39+0400 lvl=warn msg="failed to dial address, but may connect later" addr=http://localhost:9545

7|op-proposer  | t=2024-09-17T02:09:41+0400 lvl=warn msg="failed to dial address, but may connect later" addr=http://localhost:9545
```

op-batcher

```
6|op-batch | t=2024-09-17T02:09:55+0400 lvl=info msg="Initializing Batch Submitter"
6|op-batch | t=2024-09-17T02:09:55+0400 lvl=warn msg="failed to dial address, but may connect later" addr=http://localhost:9545
6|op-batch | t=2024-09-17T02:09:57+0400 lvl=warn msg="failed to dial address, but may connect later" addr=http://localhost:9545
6|op-batch | t=2024-09-17T02:09:59+0400 lvl=warn msg="failed to dial address, but may connect later" addr=http://localhost:9545
6|op-batch | t=2024-09-17T02:10:01+0400 lvl=warn msg="failed to dial address, but may connect later" addr=http://localhost:9545
6|op-batch | t=2024-09-17T02:10:03+0400 lvl=warn msg="failed to dial address, but may connect later" addr=http://localhost:9545
```

**Comment by hash1go** [comment]
Hello!
Currently, there's no standard guide for building an L3 on Base. However, in my opinion, there are two key points to consider:

- Since Base doesn't support blob transactions, you'll need to use Calldata or AltDA for data availability.
- OP Stack requires a shorter block time than the settlement layer. You'll need to hack the OP Stack block time.

For experimental approaches, you might find the unofficial "[OP Stack Hacks](https://docs.optimism.io/builders/chain-operators/hacks/overview)" guide helpful.

**Comment by dindonero** [comment]
@askucher did you manage to deploy it?

**Reply by askucher** [reply]
Nope. Do you have some scripts which can help to launch it on Base seamlessly?

**Reply by dindonero** [reply]
No, I did managed to deploy it to the point where the batcher seemed to be correctly commiting transactions to optimism sepolia (used optimism instead of base).

I got 2 errors I couldn't fix:

- Transfering ETH to the bridge contract on optimism sepolia got me logs from the op-node seeming that everything was working correctly but my balance on the L3 never got updated.
- op-proposer only started outputting logs after 2 hours of running the chain, when it tried to submit it failed and then everything started failing. The op-node started constantly producing blocks on op-geth every second, not sure why.

Anyway now I'm trying to use Arbitrum orbit. Not only their sdk is defaulted to arbitrum sepolia, but they claim in the docs that the sdk can be used for production (even tho that their github repo says otherwise).

**Reply by askucher** [reply]
I have suck issues. I have fixed them by using stable l1 rpc
